src/index.ts

```diff
@@ -210,31 +210,29 @@ export const staticPlugin = async <Prefix extends string = '/prefix'>(
         }
     })
 
+    const assetsDir = assets[0] === sep ? assets : resolve() + sep + assets
+
     if (
         alwaysStatic ||
         (process.env.ENV === 'production' && files.length <= staticLimit)
     )
-        for (let i = 0; i < files.length; i++) {
-            const filePath = files[i]
-            if (!filePath || shouldIgnore(filePath)) continue
-
-            let fileName = filePath
-                .replace(resolve(), '')
-                .replace(`${assets}${sep}`, '')
+        for (const absolutePath of files) {
+            if (!absolutePath || shouldIgnore(absolutePath)) continue
+            let relativePath = absolutePath.replace(assetsDir, '')
 
             if (noExtension) {
-                const temp = fileName.split('.')
+                const temp = relativePath.split('.')
```

Comment:
I've renamed fileName to relativePath, as this var can contain paths like `dir/subdir/file.ext`. Then I've renamed filePath to absolutePath, since they're both paths to the same file.